xlators/mgmt/glusterd/src/glusterd-syncop.c

```diff
@@ -217,6 +217,8 @@ gd_syncop_submit_request(struct rpc_clnt *rpc, void *req, void *local,
                           iobref, frame, NULL, 0, NULL, 0, NULL);
 
     /* TODO: do we need to start ping also? */
+    /* In case of error the frame will be destroy by rpc_clnt_submit */
+    frame = NULL;
```

Comment:
Same here. `rpc_clnt_submit()` may return an error without executing the callback, so the frame would leak in this case.

Reply:
You are right. I was looking at an older code version.